components/Autocomplete.tsx

```diff
@@ -0,0 +1,184 @@
+import { FC, useState, useEffect, useRef, ReactNode } from 'react'
+
+import {
+  Box,
+  Flex,
+  FlexProps,
+  Input,
+  InputProps as IProps,
+  Popover,
+  PopoverBody,
+  PopoverContent,
+  PopoverTrigger,
+  useDisclosure,
+  useMultiStyleConfig,
+} from '@chakra-ui/react'
+import { TriangleDownIcon } from '@chakra-ui/icons'
+import blem from 'blem'
+import useOutsideClickHandler from 'hooks/useOutsideClickHandler'
+
+const bem = blem('autocomplete')
+
+type OptionType = {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  value: any
+  label: string | ReactNode
+  helperText: string | ReactNode
+}
+
+interface AutocompleteProps extends FlexProps {
+  label: string
+  value?: OptionType
+  InputProps?: IProps
+  options?: OptionType[]
+  emptyOption?: ReactNode
+  renderOption?: (option: OptionType) => ReactNode
+  renderSelected?: (option: OptionType) => ReactNode
+  filterOption?: (option: OptionType, searchTerm: string) => boolean
+  onSelectOption?: (option: OptionType) => void
+}
+
+const Autocomplete: FC<AutocompleteProps> = ({
+  label,
+  value = null,
+  InputProps = {},
+  options = [],
+  emptyOption = 'No results',
+  renderOption,
+  renderSelected,
+  onSelectOption = () => void 0,
+  filterOption,
+  ...props
+}) => {
+  const [val, setVal] = useState<OptionType | null>(value)
+  const [search, setSearch] = useState<string>('')
+  const styles = useMultiStyleConfig('Autocomplete', props)
+  const inputRef = useRef<HTMLInputElement>(null)
+  const popoverRef = useRef<HTMLDivElement>(null)
+  const { onOpen, onClose, isOpen } = useDisclosure()
+
+  useOutsideClickHandler([inputRef, popoverRef], onClose)
+  useEffect(() => {
+    setVal(value)
+  }, [value])
+
+  const renderDefaultOption = option => (
+    <Box sx={styles?.option}>
+      <Box className={bem('option-label')}>{option?.label}</Box>
```

Comment:
Will `option?.label` ever be undefined? If so we should be checking for it to exist before trying to render it. If not, we should skip the optional chaining `?.`